models/tasks.js

```diff
@@ -281,6 +283,15 @@ const fetchActiveTaskMembers = async () => {
 const fetchTask = async (taskId) => {
   try {
     const task = await tasksModel.doc(taskId).get();
+    const extensionRequestSnapshot = await extensionRequestsModel.where("taskId", "==", taskId).get();
+    const extensionRequestsData = extensionRequestSnapshot.docs.map((doc) => {
+      const extensionRequestId = doc.id;
+      const status = doc.get("status");
+      return { extensionRequestId, status};
+    });
+    const isExtensionRequestPending = Boolean(
+      extensionRequestsData.filter((doc) => doc.status === EXTENSION_REQUEST_STATUS.PENDING).length
+    );
```

Comment:
Not sure of this approach, can't we just use` /extension-requests` route get the pending extension request? 

In this approach say `/task` is being 5 different places. We will be fetching and sending extension request details even thought its not required.

Reply:
@Ajeyakrishna-k Didn't understand your concern here. 
Do you want me to call  /extension-requests when tasks/:id route is hit on ui, then get this value?

Reply:
> Do you want me to call /extension-requests when tasks/:id route is hit on ui, then get this value?
I would go with this approach. I have mentioned my reasons for the same, feel free to ask others opinion.

If you want to stick to your current approach, you can add filters to query only `PENDING` extension requests and also fetch only the count.
https://firebase.google.com/docs/firestore/query-data/aggregation-queries#use_the_count_aggregation

Reply:
done. @Ajeyakrishna-k Can you please review once?